components/normalizer/src/uts46.rs

```diff
@@ -0,0 +1,150 @@
+    ///    which itself is a disallowed character.
+    /// 4. The _deviation_ characters are treated as _mapped_ or _valid_
+    ///    as appropriate.
+    /// 5. The _disallowed_STD3_valid_ characters are treated as allowed.
+    /// 6. The _disallowed_STD3_mapped_ characters are treated as
+    ///    _mapped_.
+    /// 7. The result is normalized to NFC.
+    ///
+    /// Notably:
+    ///
+    /// * The STD3 or WHATWG ASCII deny list should be implemented as a
+    ///   post-processing step.
+    /// * Transitional processing is not performed. Transitional mapping
+    ///   would be a pre-processing step, but transitional processing is
+    ///   deprecated, and none of Firefox, Safari, or Chrome use it.
+    pub fn map_normalize<'delegate, I: Iterator<Item = char> + 'delegate>(
+        &'delegate self,
+        iter: I,
+    ) -> impl Iterator<Item = char> + 'delegate {
+        self.normalizer
+            .normalize_iter_private(iter, crate::IgnorableBehavior::Ignored)
+    }
+
+    /// Returns an iterator adaptor that turns an `Iterator` over `char`
+    /// into an iterator yielding a `char` sequence that gets the following
+    /// operations from the NFC check and statucs steps of the "Validity
+    /// Criteria" section of UTS 46 lazily applied to it:
+    ///
+    /// 0. The characters from <https://www.unicode.org/versions/corrigendum4.html>
+    ///    are treated as _disallowed_.
+    /// 1. The _ignored_ characters are treated as _disallowed_.
+    /// 2. The _mapped_ characters are mapped.
+    /// 3. The _disallowed_ characters are replaced with U+FFFD,
+    ///    which itself is a disallowed character.
+    /// 4. The _deviation_ characters are treated as _mapped_ or _valid_
+    ///    as appropriate.
+    /// 5. The _disallowed_STD3_valid_ characters are treated as allowed.
+    /// 6. The _disallowed_STD3_mapped_ characters are treated as
+    ///    _mapped_.
+    /// 7. The result is normalized to NFC.
+    ///
+    /// Notably:
+    ///
+    /// * The STD3 or WHATWG ASCII deny list should be implemented as a
+    ///   post-processing step.
+    /// * Transitional processing is not performed. Transitional mapping
+    ///   would be a pre-processing step, but transitional processing is
+    ///   deprecated, and none of Firefox, Safari, or Chrome use it.
+    /// * The output needs to be compared with input to see if anything
+    ///   changed. This check catches failures to adhere to the normalization
+    ///   and status requirements.
+    pub fn normalize_validate<'delegate, I: Iterator<Item = char> + 'delegate>(
+        &'delegate self,
+        iter: I,
+    ) -> impl Iterator<Item = char> + 'delegate {
+        self.normalizer.normalize_iter_private(
+            iter.map(|c| {
+                if crate::in_inclusive_range(c, '\u{2F868}', '\u{2F9BF}') {
+                    // https://unicode-org.atlassian.net/browse/ICU-22658
+                    // Perhaps remove this special case if the PAG so decides.
```

Comment:
@eggrobin , are you able to / comfortable with guessing at this point which direction the PAG is going with this and opining on whether the special case should remain in this PR?

Reply:
It [seems to me](https://unicode-org.atlassian.net/browse/ICU-22658?focusedCommentId=174822) this special case isn't even needed. Am I missing something?

Reply:
Apologies for the delayed response, I was on vacation.

On where PAG is going: first note that PAG does not decide anything, PAG recommends and the UTC decides.
The planned PAG recommendation includes changing U+2F868, U+2F874, U+2F91F, U+2F95F, and U+2F9BF from disallowed to mapped, but includes other changes as well, trimming the base exclusion set down to 129 characters. The PAG report with these recommendations should be published shortly.

However, there is no decision until UTC #&zwnj;179, on the 23rd to 25th of this month; the UTC does not always follow WG recommendations.

On the implications on your implementation: I do not understand UTS #&zwnj;46 enough to say anything useful, ask @markusicu.

Reply:
See [L2/24-064](https://www.unicode.org/cgi-bin/GetDocumentLink?L2/24-064) Section 6.2.

Reply:
Thank you. Removing the special case.